experimental/packages/opentelemetry-instrumentation-http/src/utils.ts

```diff
@@ -676,12 +794,8 @@ export const getIncomingRequestAttributes = (
   const userAgent = headers['user-agent'];
   const ips = headers['x-forwarded-for'];
   const httpVersion = request.httpVersion;
-  const requestUrl = request.url ? url.parse(request.url) : null;
-  const host = requestUrl?.host || headers.host;
-  const hostname =
-    requestUrl?.hostname ||
-    host?.replace(/^(.*)(:[0-9]{1,5})/, '$1') ||
-    'localhost';
+  const host = headers.host;
```

Comment:
Is the host header guaranteed to be there? what about previous check to get the host from the URL?, it is not there anymore for host and hostname calculation, maybe I'm missing something still going through the changes

Reply:
AFAIK, `IncomingMessage.url` never includes the host or protocol. (https://nodejs.org/en/learn/modules/anatomy-of-an-http-transaction#method-url-and-headers, the [actual API docs don't seem to state that though](https://nodejs.org/docs/latest-v20.x/api/http.html#messageurl)) 

Under the assumption that this is true, I derived that in the old implementation - when parsing the URL with `url.parse` on L679 - there can never be a hostname. On L680 of the old code, we'll therefore always fall back to `IncomingMessage.headers.host` because the `requestUrl.hostname` it's always `null`.

For the new implementation that means: we don't try to parse the URL to get the host because it'll never be there, our only chance to get it is the host header. If it's not there, we can fall back to `localhost` as the old code did.